schemas/registry.yml

```diff
@@ -0,0 +1,74 @@
+---
+- name: registry
+  title: Registry
+  group: 2
+  description: Fields related to Windows Registry operations.
+  type: group
+  fields:
+
+    - name: hive
+      level: core
+      type: keyword
+      description: Abbreviated name for the hive.
+      example: HKLM
+
+    - name: key
+      level: core
+      type: keyword
+      description: Hive-relative path of keys.
+      example: SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\winword.exe
+
+    - name: value
+      level: core
+      type: keyword
+      example: Debugger
+      description: Name of the value written.
+
+    - name: path
+      level: core
+      type: keyword
+      description: Full path, including hive, key and value
+      example: HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\winword.exe\Debugger
+
+    - name: data.type
+      level: extended
+      type: keyword
+      description: Standard registry type for encoding contents
+      example: REG_SZ
+
+    - name: data.strings
+      level: extended
+      type: keyword
+      short: Content when writing string types.
+      example: '["C:\rta\red_ttp\bin\myapp.exe"]'
+      description: >
+        Content when writing string types.
+
+        Populated as an array when writing string data to the registry, such as REG_SZ, REG_EXPAND_SZ, REG_MULTI_SZ, and REG_LINK.
+
+    - name: data.integer
+      level: extended
+      type: long
+      short: Contents for numeric values written to the registry
+      example: 0
+      description: >
+        Contents for numeric values written to the registry
+
+        Contains the data in integer form when populating REG_DWORD and REG_QWORD. This assumes that the bytes have
+        already been interpreted in numeric form accordingly. Note that `long` is signed 64 bits, so values greater
+        than 2^63^ are invalid, and may need to be cast as negative.
+
+
+    - name: data.bytes
+      level: extended
+      type: keyword
+      short: Original bytes written with base64 encoding.
+      example: 0
+      description: >
+        Original bytes written with base64 encoding.
+
+        For Windows registry operations, such as SetValueEx and RegQueryValueEx,
+        this corresponds to the data pointed by `lp_data`. This is optional and may be redundant
+        when `data.integer` or data.strings` are populated, but should be populated
+        for REG_BINARY encoded values.
```

Comment:
I would have preferred to establish this as a value that's always reliably populated (not optional), and appropriate registry types duplicate the value in a more consumable fashion (.integer for REG_DWORD and REG_QWORD, same for strings).

However I don't see this as a blocker for the PR. Reinforcing the requirement in this direction will not be a breaking change.

What do you think? If you're not convinced of the value we can revisit this later and merge this.

Reply:
Let's revisit this later. Resolving.